plugins/BEdita/Core/tests/TestCase/Model/Entity/DateRangeTest.php

```diff
@@ -0,0 +1,652 @@
+                ],
+            ],
+            [
+                true,
+                [
+                    'start_date' => '2017-01-01',
+                    'end_date' => '2017-02-01',
+                ],
+                [
+                    'start_date' => '2017-02-01',
+                    'end_date' => null,
+                ],
+            ],
+            [
+                true,
+                [
+                    'start_date' => '2017-01-01',
+                    'end_date' => null,
+                ],
+                [
+                    'start_date' => '2017-02-01',
+                    'end_date' => '2017-02-10',
+                ],
+            ],
+            [
+                true,
+                [
+                    'start_date' => '2017-01-01',
+                    'end_date' => null,
+                ],
+                [
+                    'start_date' => '2017-02-01',
+                    'end_date' => null,
+                ],
+            ],
+            [
+                false,
+                [
+                    'start_date' => '2017-02-01',
+                    'end_date' => '2017-02-10',
+                ],
+                [
+                    'start_date' => '2017-01-01',
+                    'end_date' => '2017-01-10',
+                ],
+            ],
+            [
+                false,
+                [
+                    'start_date' => '2017-01-01',
+                    'end_date' => '2017-01-10',
+                ],
+                [
+                    'start_date' => '2017-01-01',
+                    'end_date' => '2017-01-10',
+                ],
+            ],
+            [
+                false,
+                [
+                    'start_date' => '2017-01-01',
+                    'end_date' => '2017-01-10',
+                ],
+                [
+                    'start_date' => '2017-01-05',
+                    'end_date' => null,
+                ],
+            ],
+            [
+                false,
+                [
+                    'start_date' => '2017-01-05',
+                    'end_date' => null,
+                ],
+                [
+                    'start_date' => '2017-01-01',
+                    'end_date' => '2017-01-10',
+                ],
+            ],
+            [
+                false,
+                [
+                    'start_date' => '2017-01-05',
+                    'end_date' => null,
+                ],
+                [
+                    'start_date' => '2017-01-01',
+                    'end_date' => null,
+                ],
+            ],
+            [
+                false,
+                [
+                    'start_date' => '2017-01-05',
+                    'end_date' => null,
+                ],
+                [
+                    'start_date' => '2017-01-05',
+                    'end_date' => null,
+                ],
+            ],
+        ];
```

Comment:
Tests with invalid inputs have been added for newly introduced `DateRange::checkWellFormed()` method, thus they should now be redundant here. 🤔  Do you think it would be better to add them anyway?